nexus/src/app/rack.rs

```diff
@@ -304,139 +374,140 @@ impl super::Nexus {
                 })?;
 
-            if self
-                .loopback_address_lookup(
-                    &opctx,
-                    rack_id,
-                    switch_location.into(),
-                    ipnetwork::IpNetwork::new(
-                        loopback_address_params.address,
-                        loopback_address_params.mask,
-                    )
-                    .map_err(|_| {
-                        Error::invalid_request("invalid loopback address")
-                    })?
-                    .into(),
-                )?
-                .lookup_for(authz::Action::Read)
-                .await
-                .is_err()
-            {
-                self.db_datastore
-                    .loopback_address_create(
-                        opctx,
-                        &loopback_address_params,
-                        None,
-                        &authz_address_lot,
-                    )
-                    .await?;
-            }
-
-            let name = Name::from_str("default-uplink").map_err(|e| {
-                Error::internal_error(&format!(
-                    "unable to use `default-uplink` as `Name`: {e}"
-                ))
-            })?;
-
-            let identity = IdentityMetadataCreateParams {
-                name: name.clone(),
-                description: "initial uplink configuration".to_string(),
-            };
-
-            let port_config = SwitchPortConfig {
-                geometry:
-                    nexus_types::external_api::params::SwitchPortGeometry::Qsfp28x1,
-            };
-
-            let mut port_settings_params = SwitchPortSettingsCreate {
-                identity,
-                port_config,
-                groups: vec![],
-                links: HashMap::new(),
-                interfaces: HashMap::new(),
-                routes: HashMap::new(),
-                bgp_peers: HashMap::new(),
-                addresses: HashMap::new(),
-            };
+                if self
+                    .loopback_address_lookup(
+                        &opctx,
+                        rack_id,
+                        switch_location.into(),
+                        ipnetwork::IpNetwork::new(
+                            loopback_address_params.address,
+                            loopback_address_params.mask,
+                        )
+                        .map_err(|_| {
+                            Error::invalid_request("invalid loopback address")
+                        })?
+                        .into(),
+                    )?
+                    .lookup_for(authz::Action::Read)
+                    .await
+                    .is_err()
+                {
+                    self.db_datastore
+                        .loopback_address_create(
+                            opctx,
+                            &loopback_address_params,
+                            None,
+                            &authz_address_lot,
```

Comment:
Loopback addresses can also be assigned by customers. This will be common with BGP configs and may also be used in some static routing configurations. This is why they are tracked by address lots.

For comingling, yes, system-allocated addresses should be in their own lot.

Reply:
https://github.com/oxidecomputer/omicron/issues/3603